_migrations/migration-phases/verifying-tools-for-migration.md

```diff
@@ -0,0 +1,196 @@
+
+### Verifying the S3 Repository plugin configuration
+
+You can verify that the S3 Repository Plugin is configured correctly by creating a test snapshot.
+
+Create an S3 bucket for the snapshot using the following AWS CLI command:
+
+```shell
+aws s3api create-bucket --bucket <your-bucket-name> --region <your-aws-region>
+```
+
+Register a new S3 Snapshot Repository on your source cluster using this curl command:
+
+```shell
+curl -X PUT "http://<your-source-cluster>:9200/_snapshot/test_s3_repository" -H "Content-Type: application/json" -d '{
+  "type": "s3",
+  "settings": {
+    "bucket": "<your-bucket-name>",
+    "region": "<your-aws-region>"
+  }
+}'
+```
+
+Next, create a test snapshot that captures only the cluster's metadata:
+
+```shell
+curl -X PUT "http://<your-source-cluster>:9200/_snapshot/test_s3_repository/test_snapshot_1" -H "Content-Type: application/json" -d '{
+  "indices": "",
+  "ignore_unavailable": true,
+  "include_global_state": true
+}'
+```
+
+Check the AWS Console to confirm that your bucket contains the snapshot. 
+
+### Removing test snapshots after verification
+
+To remove the resources created during verification you can use the following deletion commands:
+
+**Test snapshot**
+
+```shell
+curl -X DELETE "http://<your-source-cluster>:9200/_snapshot/test_s3_repository/test_snapshot_1?pretty"
+```
+
+**Test snapshot repository**
+
+```shell
+curl -X DELETE "http://<your-source-cluster>:9200/_snapshot/test_s3_repository?pretty"
+```
+
+**S3 bucket**
+
+```shell
+aws s3 rm s3://<your-bucket-name> --recursive
+aws s3api delete-bucket --bucket <your-bucket-name> --region <your-aws-region>
+```
+
+### Troubleshooting
+
+Use this guidance to troubleshoot any of the following snapshot verification issues:
+
+#### Access denied error (403)
+
+If you encounter an error like `AccessDenied (Service: Amazon S3; Status Code: 403)`, verify the following:
+
+- The IAM role assigned to your Elasticsearch cluster has the necessary S3 permissions.
+- The bucket name and region provided in the snapshot configuration match the actual S3 bucket you created.
+
+#### Older versions of Elasticsearch
+
+Older versions of the Elasticsearch S3 Repository Plugin may have trouble reading IAM Role credentials embedded in AWS EC2 and AWS ECS Instances. This is due to the copy of the AWS SDK shipped in them being too old to read the new standard way of retrieving those credentials, as shown in [the Instance Metadata Service v2 (IMDSv2) specification](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html). This can result in snapshot creation failures, with an error message like:
```

Comment:
```suggestion
Older versions of the Elasticsearch S3 Repository Plugin may have trouble reading IAM Role credentials embedded in Amazon EC2 and AWS ECS Instances. This is due to the copy of the AWS SDK shipped in them being too old to read the new standard way of retrieving those credentials, as shown in [the Instance Metadata Service v2 (IMDSv2) specification](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html). This can result in snapshot creation failures, with an error message like:
```